lms/djangoapps/course_api/api.py

```diff
@@ -0,0 +1,58 @@
+"""
+Course API
+"""
+
+from django.contrib.auth.models import User
+from lms.djangoapps.courseware.courses import (
+    get_courses,
+    get_course_by_id
+)
+from opaque_keys.edx.keys import CourseKey
+
+from .serializers import CourseSerializer
+
+
+def _has_permission(requesting_user, username):
+    """
+    Returns true if `requesting_user` has permission to access the user
+    identified by `username`.
+
+    Arguments:
+        requesting_user (User): The user requesting permission to view another
+        username (string): The name of the user `requesting_user` would like
+        to access.
+
+    Return value:
+        Boolean
+    """
+    if not username:
+        return False
+    return requesting_user.is_staff or requesting_user.username == username
+
+
+def course_detail(course_key_string, request):
+    """
+    Return a single course
+    """
+    # course_key = CourseKey.from_string(course_key_string)
+    # course_usage_key = modulestore().make_course_usage_key(course_key)
+
+    course = get_course_by_id(CourseKey.from_string(course_key_string))
```

Comment:
The `course_detail` method also needs to check for 'see_exists' access to the course.  The `list_courses` method is [checking access through the get_courses method](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/courses.py#L422) call.  This method also needs to do the same.  It may involve some refactoring of the `get_courses` implementation.